src/Build/BackEnd/Components/SdkResolution/SdkResolverLoader.cs

```diff
@@ -73,5 +57,58 @@ internal virtual IList<string> FindPotentialSdkResolvers(string rootFolder)
                 .Where(FileUtilities.FileExistsNoThrow)
                 .ToList();
         }
+
+        protected virtual IEnumerable<Type> GetResolverTypes(Assembly assembly)
+        {
+            return assembly.ExportedTypes
+                .Select(type => new {type, info = type.GetTypeInfo()})
+                .Where(t => t.info.IsClass && t.info.IsPublic && !t.info.IsAbstract && typeof(SdkResolver).IsAssignableFrom(t.type))
+                .Select(t => t.type);
+        }
+
+        protected virtual Assembly LoadResolverAssembly(string resolverPath, LoggingContext loggingContext, ElementLocation location)
+        {
+#if FEATURE_ASSEMBLY_LOADFROM
+            return Assembly.LoadFrom(resolverPath);
+#else
+            _loader.AddDependencyLocation(Path.GetDirectoryName(resolverPath));
+            return _loader.LoadFromPath(resolverPath);
+#endif
+        }
+
+        protected virtual void LoadResolvers(string resolverPath, LoggingContext loggingContext, ElementLocation location, List<SdkResolver> resolvers)
+        {
+            Assembly assembly;
+            try
+            {
+                assembly = LoadResolverAssembly(resolverPath, loggingContext, location);
+            }
+            catch (Exception e)
+            {
+                loggingContext.LogWarning(null, new BuildEventFileInfo(location), "CouldNotLoadSdkResolverAssembly", resolverPath, e.Message);
```

Comment:
Should this be an error or warning? If one of the resolver fails to load it would probably mean your MSBuild instance is messed up. You could end up hitting a different resolver that may work.

Reply:
It is now throwing an `InvalidProjectFileException` which stops the build with a logged error or can be caught by consumers of the API.